build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchTestBasePlugin.java

```diff
@@ -199,5 +200,24 @@ public void execute(Task t) {
                 }
             });
         });
+        configureImmutableCollectionsPatch(project);
+    }
+
+    private void configureImmutableCollectionsPatch(Project project) {
+        String configurationName = "immutableCollectionsPatch";
+        var patchConfiguration = project.getConfigurations().register(configurationName, config -> {
+            config.setCanBeConsumed(false);
+        });
```

Comment:
```suggestion
        FileCollection patchedFileCollection = project.getConfigurations().register(configurationName, config -> {
            config.setCanBeConsumed(false);
        }).get();
```

Reply:
what you create with 'register' is a `NamedDomainObjectProvider<Configuration>` and we need the configuration itself that's why we use get() to get the actual `Configuration` instance. its more or less the same as doing `configurations.create(...)` in the first place.  Here it doesn't make a difference. In general gradle added this register api to make creation of certain domain objects lazy which it considers expensive to create and maybe never used. Its similar to avoid creating tasks that are not used by using tasks.register()

Reply:
Let's just use `create()` instead of `register()` then if we don't actually need a `Provider`.